JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -480,4 +831,69 @@ private List<String> getInstances(SelectQuery instanceSelectQuery, String placeh
 
 		return instanceIRIs;
 	}
+
+	protected void initTS(String timeSeriesIRI, List<String> dataIRI, String dbURL, String timeUnit, Iri type, String durationIRI, Double duration, String unit){
+		// Construct time series IRI
+		Iri tsIRI;
+		// Check whether timeseriesIRI follows IRI naming convention of namespace & local_name
+		// If only local name is provided, iri() function would add filepath as default prefix
+		// Every legal (full) IRI contains at least one ':' character to separate the scheme from the rest of the IRI
+		if (Pattern.compile("\\w+\\S+:\\S+\\w+").matcher(timeSeriesIRI).matches()) {
+			tsIRI = iri(timeSeriesIRI);
+		} else {
+			throw new JPSRuntimeException(exceptionPrefix + "Time series IRI does not have valid IRI format");
+		}
+
+		ModifyQuery modify = Queries.MODIFY();
+		// set prefix declarations
+		modify.prefix(prefix_ontology, prefix_kb, prefix_time);
+
+		// define type
+		modify.insert(tsIRI.isA(type));
+
+		if(type.equals(AverageTimeSeries)){
+
+			//numeric Duration
+			if(duration <=0.0){
+				throw new JPSRuntimeException(exceptionPrefix + "Numeric Duration must be a positive value");
+			}
+
```

Comment:
see previous comment

Reply:
this method deals with double duration and not the duration object

Reply:
Since you already make a check before uploading this I don't think this is necessary